base/parse.jl

```diff
@@ -58,6 +58,10 @@ end
 
 function tryparse_internal{T<:Integer}(::Type{T}, s::AbstractString, startpos::Int, endpos::Int, base_::Integer, raise::Bool)
     _n = Nullable{T}()
+    if isempty(strip(s))
```

Comment:
I think calling `strip` will be too expensive here --- this function is very performance-critical, and most inputs don't need `strip` at all. I believe `parseint_preamble` returns a base of `0` if there is a problem; we should check for all-whitespace strings only in that case.